Realm/RLMSyncConfiguration.mm

```diff
@@ -56,6 +61,50 @@ RLMSyncSystemErrorKind errorKindForSyncError(SyncError error) {
 }
 }
 
+struct BeforeClientResetWrapper {
+    RLMClientResetBeforeBlock block;
+    RLMRealmConfiguration *rlmConfig;
```

Comment:
This doesn't appear to need the entire RLMRealmConfiguration? Storing just the path and custom schema would fix the retain cycle and reduce memory usage.